src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -1004,7 +1009,40 @@ internal static IEnumerable<EnvDTE.Project> GetDependentEnvDTEProjects(IDictiona
             dependentEnvDTEProjects.Add(dependentEnvDTEProject);
         }
 
-#region IVsSelectionEvents
+        /// <summary>
+        /// This method is invoked when ProjectSystemCache fires a CacheUpdated event.
+        /// This method inturn invokes AfterNuGetCacheUpdated event which is consumed by PackageManagerControl.xaml.cs
+        /// </summary>
+        /// <param name="sender">Event sender object</param>
+        /// <param name="e">Event arguments. This will be EventArgs.Empty</param>
+        private void NuGetCacheUpdate_After(object sender, EventArgs e)
+        {
+            // The AfterNuGetCacheUpdated event is raised on a separate Task to prevent blocking of the caller.
+            // E.g. - If Restore updates the cache entries on CPS nomination, then restore should not be blocked till UI is restored.
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(() => FireNuGetCacheUpdatedEventAsync(e));
```

Comment:
And if not used properly then JTF is also the root for any deadlocks in VS, what do you see wrong with Task.Run as in this case it's like fire and forget about this task so don't you think it will be a better approach?